### ISSUE TITLE
Another ArgumentError

### ISSUE THREAD

**Issue by bradrobertson** [nylas/nylas-ruby#301]
I believe this is similar to #257. It's hard to tell exactly how to reproduce this due to the way the gem is written. But _some_ API responses (I'm assuming when an error occurs) don't return the expected object which is passed directly into the constructor that doesn't conform to the interface of that constructor. Here's a stack trace:

```
Unable to fetch deltas due to /usr/local/bundle/gems/nylas-4.6.5/lib/nylas/model/transferable.rb:12:in `initialize': wrong number of arguments (given 1, expected 0) (ArgumentError)
	from /usr/local/bundle/gems/nylas-4.6.5/lib/nylas/deltas_collection.rb:36:in `new'
	from /usr/local/bundle/gems/nylas-4.6.5/lib/nylas/deltas_collection.rb:36:in `execute'
	from /usr/local/lib/ruby/2.6.0/forwardable.rb:224:in `cursor_end'
```

the offending line appears to be here: `self.deltas ||= Deltas.new(api.execute(to_be_executed))` so something returned by `api.execute` doesn't conform to the `Transferable` constructor definition of `def initialize(**initial_data)`

Again, I assume you can only reproduce this during some error state

**Comment by mrashed-dev** [comment]
Thanks for bringing this to our attention, we're looking into it!

**Comment by arunagw** [comment]
Hello @bradrobertson thanks for reporting this. Seems like you are on version 4.6.5. We pushed a fix for this in this commit https://github.com/nylas/nylas-ruby/commit/4669fba87746e98ce9f3d49f56ebd86572db6d80 and it's available in latest version of gem. Can you please try upgrading to 4.6.7 and see if this problem still exists? 

We also had a proper fix for Delta module in #297 but it hasn't been released yet. Will try to get it out very soon.

**Comment by arunagw** [comment]
Hey @bradrobertson did you get a chance to get it verified with latest version of gem? I have check and it seems to be working fine. 
Closing this issue for now comment back or reopen this if issue still exists. 🙏
